base/applications/mspaint/mouse.cpp

```diff
@@ -480,6 +478,7 @@ struct SmoothDrawTool : ToolBase
     void OnFinishDraw() override
     {
         ToolBase::OnFinishDraw();
+        m_bShiftDown = FALSE;
```

Comment:
Should this be done here? Or in the OnButtonUp function? (since m_bShiftDown is set to TRUE/FALSE in OnButtonDown)

Reply:
This statement is needless.

```suggestion
```

Reply:
Can you explain why is it useless? Shouldn't the shift-state be reset to its default (FALSE) value? Or is it because the draw-tool object dies out just after the OnFinishDraw() function gets called?

Reply:
The drawing stroke is finished at here. This flag won't be used until the next stroke does come.

Reply:
ok thanks for the explanation